src/main/java/nz/cri/gns/NSHM/opensha/ruptures/NSHMRuptureSetBuilder.java

```diff
@@ -0,0 +1,265 @@
+package nz.cri.gns.NSHM.opensha.ruptures;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.dom4j.DocumentException;
+import org.opensha.refFaultParamDb.vo.FaultSectionPrefData;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.ClusterRupture;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.ClusterRuptureBuilder;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.FaultSubsectionCluster;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.plausibility.PlausibilityConfiguration;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.plausibility.impl.CumulativeAzimuthChangeFilter;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.plausibility.impl.JumpAzimuthChangeFilter;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.plausibility.impl.MinSectsPerParentFilter;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.plausibility.impl.TotalAzimuthChangeFilter;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.strategies.ClusterConnectionStrategy;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.strategies.ClusterPermutationStrategy;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.strategies.ConnectionPointsPermutationStrategy;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.strategies.DistCutoffClosestSectClusterConnectionStrategy;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.strategies.UCERF3ClusterPermuationStrategy;
+import org.opensha.sha.earthquake.faultSysSolution.ruptures.util.SectionDistanceAzimuthCalculator;
+import org.opensha.sha.faultSurface.FaultSection;
+
+import com.google.common.base.Preconditions;
+
+import nz.cri.gns.NSHM.opensha.ruptures.downDipSubSectTest.DownDipSubSectBuilder;
+import nz.cri.gns.NSHM.opensha.ruptures.downDipSubSectTest.DownDipTestPermutationStrategy;
+import nz.cri.gns.NSHM.opensha.ruptures.downDipSubSectTest.SubSectionParentFilter;
+import nz.cri.gns.NSHM.opensha.util.FaultSectionList;
+import scratch.UCERF3.SlipAlongRuptureModelRupSet;
+import scratch.UCERF3.enumTreeBranches.FaultModels;
+import scratch.UCERF3.enumTreeBranches.ScalingRelationships;
+import scratch.UCERF3.enumTreeBranches.SlipAlongRuptureModels;
+
+/**
+ * Builds opensha SlipAlongRuptureModelRupSet rupture sets 
+ * using NZ NSHM configurations for:
+ *  - plausability
+ *  - rupture permutation Strategy (with different strategies available for test purposes)
+ */
+public class NSHMRuptureSetBuilder {
+
+	static DownDipSubSectBuilder downDipBuilder;	
+	Set<Integer> faultIdIn = Collections.emptySet();
+	
+	double maxSubSectionLength = 0.5; 	// maximum sub section length (in units of DDW)
+	double maxDistance = 5; 			// max distance for linking multi fault ruptures, km
```

Comment:
Odd indentation of the comment. Did you auto-format this file?

Reply:
I wonder if this is a tab-size thing? It look fine in my editor. My Eclipse tab-size is 4. But this is github ??

Reply:
@voj what's the standard tab size for java code? (In python it's 4)

Reply:
I found this https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment so I've decided to remove/stop creating that alignment style in java.